### PR TITLE
Move files_versions to bootstrap mechanism

### PR DESCRIPTION
**Pull request by MorrisJobke**

### CODE REVIEW

File: apps/files_versions/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -71,20 +70,23 @@ public function __construct(array $urlParams = []) {
...
 		/**
 		 * Register Events
 		 */
-		$this->registerEvents($dispatcher);
+		$context->registerEventListener(LoadAdditionalScriptsEvent::class, LoadAdditionalListener::class);
+		$context->registerEventListener(LoadSidebar::class, LoadSidebarListener::class);
+	}
+
+	public function boot(IBootContext $context): void {
+		$this->registerVersionBackends();
```
Comment (ChristophWurst) [inline apps/files_versions/lib/AppInfo/Application.php:89]: you could add the `IServerContainer` and `IAppManager` to the function arguments and call it with `$context->injectFn(Closure::fromCallable([$this, 'registerVersionBackends']));` here

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
> PHP Fatal error:  require_once(): Failed opening required '/drone/src/apps/files_versions/tests/../appinfo/app.php' in /drone/src/apps/files_versions/tests/VersioningTest.php on line 38

:popcorn:

File: apps/files_versions/lib/AppInfo/Application.php (hunk 1 above)
Comment (MorrisJobke) [reply apps/files_versions/lib/AppInfo/Application.php:89]: Done.

**Comment by MorrisJobke** [comment]
> > PHP Fatal error:  require_once(): Failed opening required '/drone/src/apps/files_versions/tests/../appinfo/app.php' in /drone/src/apps/files_versions/tests/VersioningTest.php on line 38
> 
> 🍿

I removed that, because all it did was to initialize the Application, but the constructor didn't executed anything. So let's see and have some 🍿

**Comment by MorrisJobke** [comment]
Worked also on CI. Thus I squashed it and it's ready for review :)

**Review by ChristophWurst — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**